core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -139,6 +189,28 @@ public void put(PutPipelineRequest request, ActionListener<IndexResponse> listen
             throw new IllegalArgumentException("Invalid pipeline configuration", e);
         }
 
+        ClusterState state = clusterService.state();
+        if (validateIngestIndex(state)) {
```

Comment:
call the method isIngestIndexPresent ? and document that it will throw exception if the index is there but not consistent?